pkg/importer/dataStream_test.go

```diff
@@ -79,239 +172,99 @@ func TestNewDataStream(t *testing.T) {
 }
 
 func Test_dataStream_Read(t *testing.T) {
-	type fields struct {
-		Url         *url.URL
-		Readers     []Reader
-		buf         []byte
-		Qemu        bool
-		Size        int64
-		accessKeyId string
-		secretKey   string
-	}
-
-	type args struct {
-		buf []byte
-	}
 	tests := []struct {
-		name    string
-		fields  fields
-		args    args
-		want    int
-		wantErr bool
+		name     string
+		testFile string
+		wantErr  bool
 	}{
-	// TODO: Add test cases.
+		{
+			name:     "successful read of datastream buffer",
+			testFile: "sample.tar.xz",
+			wantErr:  false,
+		},
+		{
+			name:     "failure on invalid read of empty datastream buffer",
+			testFile: "",
+			wantErr:  true,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			d := dataStream{
-				Url:         tt.fields.Url,
-				Readers:     tt.fields.Readers,
-				buf:         tt.fields.buf,
-				qemu:        tt.fields.Qemu,
-				Size:        tt.fields.Size,
-				accessKeyId: tt.fields.accessKeyId,
-				secretKey:   tt.fields.secretKey,
-			}
-			got, err := d.Read(tt.args.buf)
+			ds, testBytes, expectedSize := createDataStreamBuf("", "", "", tt.testFile)
+			defer ds.Close()
+
+			got, err := ds.Read(testBytes)
 			if (err != nil) != tt.wantErr {
-				t.Errorf("dataStream.Read() error = %v, wantErr %v", err, tt.wantErr)
+				t.Errorf("dataStream.Read() error = %v, wantErr %v %v %v", err, tt.wantErr, testBytes, expectedSize)
 				return
 			}
-			if got != tt.want {
-				t.Errorf("dataStream.Read() = %v, want %v", got, tt.want)
+			if got != expectedSize {
+				t.Errorf("dataStream.Read() = %v, want %v", got, expectedSize)
 			}
 		})
 	}
 }
 
 func Test_dataStream_Close(t *testing.T) {
-	type fields struct {
-		Url         *url.URL
-		Readers     []Reader
-		buf         []byte
-		qemu        bool
-		Size        int64
-		accessKeyId string
-		secretKey   string
-	}
+	imageDir, _ := filepath.Abs("../../test/images")
+	localImageBase := filepath.Join("file://", imageDir)
+	ds := createDataStreamMulti("sample.tar.xz", localImageBase)
+	defer ds.Close()
```

Comment:
the `Close` method returns an error which is not being checked here.
Also, if the ds.Close below is successful then this Close may fail (but maybe that's why you're ignoring the err here?)